File: 167-TwoSumIiInputArrayIsSorted/167-TwoSumIiInputArrayIsSorted.cpp
// Last updated: 7/29/2026, 10:13:28 AM
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // two pointers, i and j. i at front and j at back. 
        // conditionals: if the sum equal(return indices), the sum is less(increment i), the sum is greater(decrement j)
        int i = 1;
        int j = numbers.size();
        while(i != j){
            if(numbers[i - 1] + numbers[j -1] == target)
                return {i, j};
            else if(numbers[i - 1] + numbers[j -1] <= target)
                i++;
            else
                j--;
        }
        return {0};
    }
};

File: 167-TwoSumII-InputArrayIsSorted/167-TwoSumII-InputArrayIsSorted.py
# Last updated: 8/8/2026, 10:35:25 AM
1class Solution:
2    def twoSum(self, numbers: List[int], target: int) -> List[int]:
3        result = [1] * 2
4        l = 0
5        r = len(numbers) - 1
6        while l < r:
7            if numbers[l] + numbers[r] == target:
8                result[0] += l
9                result[1] += r
10                return result
11            else:
12                if numbers[l] + numbers[r] > target:
13                    r -= 1
14                else:
15                    l += 1
16        return result